sdk/tests/sdk/whirlpools/whirlpool-impl#closePosition.test.ts

```diff
@@ -0,0 +1,552 @@
+    const whirlpoolClient = buildWhirlpoolClient(whirlpoolCtx);
+
+    testCtx = {
+      provider,
+      program,
+      whirlpoolCtx,
+      whirlpoolClient,
+    };
+  });
+
+  async function accrueFees(fixture: WhirlpoolTestFixture) {
+    const ctx = testCtx.whirlpoolCtx;
+    const {
+      poolInitInfo,
+      positions: [positionInfo],
+      tokenAccountA,
+      tokenAccountB,
+    } = fixture.getInfos();
+
+    const { whirlpoolPda, tokenVaultAKeypair, tokenVaultBKeypair } = poolInitInfo;
+
+    const tickArrayPda = PDAUtil.getTickArray(ctx.program.programId, whirlpoolPda.publicKey, 22528);
+    const oraclePda = PDAUtil.getOracle(ctx.program.programId, whirlpoolPda.publicKey);
+
+    const pool = await testCtx.whirlpoolClient.getPool(whirlpoolPda.publicKey);
+    const position = await testCtx.whirlpoolClient.getPosition(positionInfo.publicKey);
+
+    // Accrue fees in token A
+    await toTx(
+      ctx,
+      WhirlpoolIx.swapIx(ctx.program, {
+        amount: new u64(200_000),
+        otherAmountThreshold: ZERO_BN,
+        sqrtPriceLimit: MathUtil.toX64(new Decimal(4)),
+        amountSpecifiedIsInput: true,
+        aToB: true,
+        whirlpool: whirlpoolPda.publicKey,
+        tokenAuthority: ctx.wallet.publicKey,
+        tokenOwnerAccountA: tokenAccountA,
+        tokenVaultA: tokenVaultAKeypair.publicKey,
+        tokenOwnerAccountB: tokenAccountB,
+        tokenVaultB: tokenVaultBKeypair.publicKey,
+        tickArray0: tickArrayPda.publicKey,
+        tickArray1: tickArrayPda.publicKey,
+        tickArray2: tickArrayPda.publicKey,
+        oracle: oraclePda.publicKey,
+      })
+    ).buildAndExecute();
+
+    // Accrue fees in token B
+    await toTx(
+      ctx,
+      WhirlpoolIx.swapIx(ctx.program, {
+        amount: new u64(200_000),
+        otherAmountThreshold: ZERO_BN,
+        sqrtPriceLimit: MathUtil.toX64(new Decimal(5)),
+        amountSpecifiedIsInput: true,
+        aToB: false,
+        whirlpool: whirlpoolPda.publicKey,
+        tokenAuthority: ctx.wallet.publicKey,
+        tokenOwnerAccountA: tokenAccountA,
+        tokenVaultA: tokenVaultAKeypair.publicKey,
+        tokenOwnerAccountB: tokenAccountB,
+        tokenVaultB: tokenVaultBKeypair.publicKey,
+        tickArray0: tickArrayPda.publicKey,
+        tickArray1: tickArrayPda.publicKey,
+        tickArray2: tickArrayPda.publicKey,
+        oracle: oraclePda.publicKey,
+      })
+    ).buildAndExecute();
+
+    const poolData = await pool.refreshData();
+    const positionData = await position.refreshData();
+    const tickLowerData = position.getLowerTickData();
+    const tickUpperData = position.getLowerTickData();
+
+    const quote = collectFeesQuote({
+      whirlpool: poolData,
+      position: positionData,
+      tickLower: tickLowerData,
+      tickUpper: tickUpperData,
+    });
+
+    assert.ok(quote.feeOwedA.gtn(0) || quote.feeOwedB.gtn(0));
+  }
+
+  async function accrueRewards(fixture: WhirlpoolTestFixture) {
```

Comment:
The quotes doesn't actually accrue rewards. The rewards are accrued when the swap occurs (in accrue fees)